src/plugins/intel_npu/src/plugin/npuw/partitioning/online/compiler.cpp

```diff
@@ -323,7 +323,17 @@ class Compiler {
         LOG_INFO("Online partitioning: compiling single group pipeline...");
         LOG_BLOCK();
 
-        m_snapshot->singleGroup();
+        // Note: Assuming the context was set with minimum graph size = 1
+        // Note: If there are avoids present there will be multiple groups formed instead of 1
+        m_snapshot->earlyAvoids();
+        m_snapshot->repeat([&] {
+            m_snapshot->repeat([&] {
+                m_snapshot->collectLHF();
+            });
+            m_snapshot->repeat([&] {
+                m_snapshot->fuseRemnantsExtended();
+            });
+        });
```

Comment:
I'd check if there's any `AVOID`s, then go hard way, otherwise `singleGroup()` could still work.

`AVOID`s are still an exception rather than somewhat normal

Reply:
Done